### PR TITLE
KEP 3756: Mark NewVolumeManagerReconstruction as GA

### PR DESCRIPTION
**Pull request by jsafrane**
#### What type of PR is this?
/kind feature
/kind cleanup

#### What this PR does / why we need it:
Mark `NewVolumeManagerReconstruction` as GA, remove all references to the feature gate and clean up the code:

* Remove all files and functions with the "old" volume reconstruction. This code is unreachable with the feature gate GA.
* Remove "`new`" suffix from all files and the function with the "new" reconstruction, there is just one reconstruction now.

It is easier to review the commits one by one, as `reconciler.go` and `reconstruct.go` (with the "old" code) are removed in one commit and then `reconciler_new.go` and `reconstruct_new.go` is renamed to `reconciler.go` and `reconstruct.go` in another commit. It shows a complicated diff when reviewing the whole PR, while there is just file removal + rename.

#### Does this PR introduce a user-facing change?

```release-note
NewVolumeManagerReconstruction feature is now GA.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3756-volume-reconstruction

```

### CODE REVIEW

**Comment by jsafrane** [comment]
cross referencing the enhancement issue: https://github.com/kubernetes/enhancements/issues/3756

**Comment by carlory** [comment]
Before promoting it to GA, could you confirm whether kubernetes/kubernetes#121937 is in consideration for robust VolumeManager reconstruction? @jsafrane 

I noticed that  @yuqitao has commented on it in the KEP issue.

File: pkg/kubelet/volumemanager/reconciler/reconciler.go [hunk 1]
```diff
@@ -14,119 +14,62 @@ See the License for the specific language governing permissions and
...
 )
 
-func (rc *reconciler) runOld(stopCh <-chan struct{}) {
-	wait.Until(rc.reconciliationLoopFunc(), rc.loopSleepDuration, stopCh)
-}
-
-func (rc *reconciler) reconciliationLoopFunc() func() {
-	return func() {
-		rc.reconcile()
+// TODO: move to reconciler.go and remove old code there when NewVolumeManagerReconstruction is GA
```
Comment (carlory) [inline pkg/kubelet/volumemanager/reconciler/reconciler.go:24]: remove this line.

File: pkg/kubelet/volumemanager/reconciler/reconciler.go [hunk 2]
```diff
@@ -14,119 +14,62 @@ See the License for the specific language governing permissions and
...
 
-		// Sync the state with the reality once after all existing pods are added to the desired state from all sources.
-		// Otherwise, the reconstruct process may clean up pods' volumes that are still in use because
-		// desired state of world does not contain a complete list of pods.
-		if rc.populatorHasAddedPods() && !rc.StatesHasBeenSynced() {
-			klog.InfoS("Reconciler: start to sync state")
-			rc.sync()
-		}
-	}
+// TODO: Replace Run() when NewVolumeManagerReconstruction is GA
```
Comment (carlory) [inline pkg/kubelet/volumemanager/reconciler/reconciler.go:26]: and here.

File: pkg/kubelet/volumemanager/reconciler/reconciler_common.go [hunk 3]
```diff
@@ -149,12 +147,7 @@ type reconciler struct {
 }
 
 func (rc *reconciler) Run(stopCh <-chan struct{}) {
-	if utilfeature.DefaultFeatureGate.Enabled(features.NewVolumeManagerReconstruction) {
-		rc.runNew(stopCh)
-		return
-	}
-
-	rc.runOld(stopCh)
+	rc.run(stopCh)
```
Comment (carlory) [inline pkg/kubelet/volumemanager/reconciler/reconciler_common.go:150]: rename `run` to `Run`

File: pkg/kubelet/volumemanager/reconciler/reconciler.go (hunk 1 above)
Comment (jsafrane) [reply pkg/kubelet/volumemanager/reconciler/reconciler.go]: Fixed

File: pkg/kubelet/volumemanager/reconciler/reconciler_common.go (hunk 3 above)
Comment (jsafrane) [reply pkg/kubelet/volumemanager/reconciler/reconciler_common.go]: Good catch, fixed

**Comment by jsafrane** [comment]
> Before promoting it to GA, could you confirm whether https://github.com/kubernetes/kubernetes/issues/121937 is in consideration for robust VolumeManager reconstruction? @jsafrane

I answered a similar question in the [enhancement issue](https://github.com/kubernetes/enhancements/issues/3756#issuecomment-1905822482). I think it should be a separate feature with a separate feature gate (or even without a feature gate, if done carefuly)

**Pull request merged**